### ISSUE TITLE
[V2] backButton only renders previous screen title when pushing on iOS

### ISSUE THREAD

**Issue by pribeh** [wix/react-native-navigation#4073]
As of commit 2.0.2569, most backButton properties don't seem to take effect when pushing a new screen. backButton properties were working fine with version 2.0.2555.

For instance, if I use the following, the title in the default blue color displays with **no** icon:

```
	static options() {
		return {
			topBar: {
				backButton: {
					icon: require('../img/navbar/back.png'),
					color:'#262626',
					showTitle: false
				},
			},
		};
	}
```

---
### Environment
* React Native Navigation version: 2.0.2569
* React Native version: 0.55.4
* Platform(s) (iOS, Android, or both?): iOS
* Device info (Simulator/Device? OS version? Debug/Release?): iPhone 6, X & simulator running iOS 12

**Comment by pribeh** [comment]
As I suspected, if I revert changes in commit [#3999](https://github.com/wix/react-native-navigation/commit/2cd264ec839f25b2be65dee42821adb6913f98c0) then the backbutton properties work again as expected.

**Comment by srameshr** [comment]
```2.0.2569``` I agree, I am on ```2.0.2569``` and none of the back button properties work.

**Comment by srameshr** [comment]
@pribeh Could you revert or fix this?

**Comment by pribeh** [comment]
There are currently a few things that 2.0.2569 seems to break including modal background properties. I'm trying to think of what might be the best thing to recommend to the maintainers to do.

It also looks like there may be different expectations as to how the backbutton ought to work from the commit [#3999](https://github.com/wix/react-native-navigation/commit/2cd264ec839f25b2be65dee42821adb6913f98c0).

**Comment by pribeh** [comment]
As of 2.0.2584 this appears to be resolved. @srameshr can you . test and confirm?

**Comment by pribeh** [comment]
Okay, so looks like there's some refactoring of the whole button code for iOS in 2.0.2584. But it does seem to introduce a build error (see attached). Might have to file this as a separate issue.

<img width="1141" alt="screen shot 2018-10-16 at 10 46 55 am" src="https://user-images.githubusercontent.com/690883/47026493-93eb2a80-d133-11e8-97af-3e05952d73ba.png">

**Comment by yogevbd** [comment]
Looking into it now. thanks!

**Comment by yogevbd** [comment]
@pribeh pushed fix. Please let me know if this version solves the back button issue. thanks

**Comment by pribeh** [comment]
I believe this is all resolved as of 2.0.2585 – including the build error. thanks @yogevbd !
